src/main/java/com/uid2/admin/salt/KeyIdGenerator.java

```diff
@@ -0,0 +1,45 @@
+package com.uid2.admin.salt;
+
+import com.uid2.shared.model.SaltEntry;
+
+import java.util.Arrays;
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class KeyIdGenerator {
```

Comment:
I think it's worthwhile to add a comment somewhere in this class about the assumptions of key IDs in the latest rotated buckets and the intended outcomes of this logic:
- Key IDs from the latest buckets were the last allocated key IDs
- These key IDs should always be consecutive
- If the last allocated key ID contains 16777215, we should wrap around
- Continuing to increment from the last "highest" key ID will result in monotonic incrementation of key IDs for all newly rotated buckets

Hopefully my understanding is correct, it took me a while to grasp all of these assumptions and outcomes so a comment would help future readers of this code